### PR TITLE
subjective classification

### PR DESCRIPTION
**Pull request by jsamoocha**
Work-in-progress on a subjective meta classifier, that uses a prior class distribution to compute posterior probabilities of predicted classes (#201 ).

### CODE REVIEW

File: sklego/meta.py [hunk 1]
```diff
@@ -537,3 +537,33 @@ def predict(self, X):
...
+
+class SubjectiveClassifier(BaseEstimator, ClassifierMixin, MetaEstimatorMixin):
+    def __init__(self, estimator, prior):
+        self.estimator = estimator
+        self.prior = prior
+
+    def fit(self, X, y):
+        X, y = check_X_y(X, y, estimator=self.estimator, dtype=FLOAT_DTYPES)
+        if not isinstance(self.estimator, ClassifierMixin):
+            raise ValueError("The SubjectiveClassifier meta model only works on classifcation models")
```
Comment (koaning) [inline sklego/meta.py]: spelling

**Comment by koaning** [comment]
Is there a reason why you did not implement `predict_proba()`?

**Comment by koaning** [comment]
I am also a little bit confused by the implementation. The way I interpreted it, the estimated probabilty (y_hat in the formula below), is given by `self.estimator.predict(X)`. The prior is p(y_a) and the confusion matrix determines p(y_hat_a | y_a).  



Did I get this idea wrong?

**Comment by jsamoocha** [comment]
> I am wondering if it is easier to have the internals be pandas dataframes ... it might be better if you keep numpy arrays because then you can multiply matrices with `@`.

There doesn't seem to be a need for matrix multiplication in this case. The `DataFrame` allows easier lookups for non-numeric classes using `index` and `columns`.

> Is there a reason why you did not implement `predict_proba()`?

TODO - this is all WIP. Just wanted to quickly share the core principle to avoid any confusion. 


> I am also a little bit confused by the implementation. The way I interpreted it, the estimated probabilty (y_hat in the formula below), is given by `self.estimator.predict(X)`. The prior is p(y_a) and the confusion matrix determines p(y_hat_a | y_a).
> 
> 
> 
> Did I get this idea wrong?

For a binary classifier, if you expand the denominator p(y_hat_a), it becomes p(y_hat_a | y_a)*p(y_a) + p(y_hat_a | y_b)*p(y_b). The `_evidence()` method computes this denominator for the general case with _n_ classes.

**Comment by koaning** [comment]
check that errors go to .fit 
use self.classes_ in favor of self.estimator.classes_
check classes attr in first test -> classes_
add standard tests
docstring

**Comment by koaning** [comment]
Merged as an experimental feature. Won't add documentation for it yet.

**Comment by koaning** [comment]
@jsamoocha thanks! I think this is an interesting idea to keep exploring. I've added it to the repo for now. Let's ping back later. I will also close all related issues.

**Pull request merged**